internal/rbd/rbd_util.go

```diff
@@ -1185,17 +1196,27 @@ func GenVolFromVolID(
 		!errors.Is(err, ErrImageNotFound) {
 		return vol, err
 	}
+	defer func() {
+		if err != nil {
+			vol.Destroy()
+		}
+	}()
 
 	// Check clusterID mapping exists
-	mapping, mErr := util.GetClusterMappingInfo(vi.ClusterID)
-	if mErr != nil {
-		return vol, mErr
+	mapping, err := util.GetClusterMappingInfo(vi.ClusterID)
+	if err != nil {
+		return vol, err
 	}
 	if mapping != nil {
-		rbdVol, vErr := generateVolumeFromMapping(ctx, mapping, volumeID, vi, cr, secrets)
-		if !errors.Is(vErr, util.ErrKeyNotFound) && !errors.Is(vErr, util.ErrPoolNotFound) &&
-			!errors.Is(vErr, ErrImageNotFound) {
-			return rbdVol, vErr
+		// create a new volume based on the mapping, cleanup the old volume
+		if vol != nil {
+			vol.Destroy()
+		}
```

Comment:
when will we hit this one? In case of error we are already doing the Destory and in case of non-nil error case the vol is returned at line 1197